tests/src/Unit/PermissionEventTest.php

```diff
@@ -380,6 +547,73 @@ public function testIteratorAggregate($permissions, $entity_type_id, $bundle_id,
   }
 
   /**
+   * Tests creation of an invalid operation permission.
+   *
+   * @expectedException \InvalidArgumentException
+   */
+  public function testInvalidGroupContentOperationPermissionCreation() {
+    // An exception should be thrown when a group content operation permission
+    // is created with an invalid ownership type.
+    new GroupContentOperationPermission([
+      'name' => 'invalid permission',
+      'title' => $this->t('This is an invalid permission.'),
+      'entity type' => 'node',
+      'bundle' => 'article',
+      'operation' => 'create',
+      'ownership' => 'an invalid ownership',
+    ]);
+  }
+
+  /**
+   * Tests that it is possible to overwrite an existing operation permission.
+   *
+   * Operation permissions are not identified by their machine name, but by the
+   * unique combination of entity type, bundle, operation and ownership. It
+   * should be possible to change the name, title and other properties.
+   */
+  public function testOverwritingGroupContentOperationPermission() {
+    $event = new PermissionEvent($this->randomMachineName(), $this->randomMachineName(), []);
+
+    $entity_type_id = $this->randomMachineName();
+    $bundle_id = $this->randomMachineName();
+    $operation = $this->randomMachineName();
+    $ownership = 'own';
+
+    $original_permission = new GroupContentOperationPermission([
+      'name' => 'the original permission',
+      'title' => $this->t('This is the original permission.'),
+      'description' => $this->t('This is the original description.'),
+      'entity type' => $entity_type_id,
+      'bundle' => $bundle_id,
+      'operation' => $operation,
+      'ownership' => $ownership,
+      'default roles' => [OgRoleInterface::ADMINISTRATOR],
+      'restrict access' => TRUE,
+    ]);
+
+    $altered_permission = new GroupContentOperationPermission([
+      'name' => 'the altered permission',
+      'title' => $this->t('This is the altered permission.'),
+      'description' => $this->t('This is the altered description.'),
+      'entity type' => $entity_type_id,
+      'bundle' => $bundle_id,
+      'operation' => $operation,
+      'ownership' => $ownership,
+      'default roles' => [OgRoleInterface::AUTHENTICATED],
+      'restrict access' => FALSE,
+    ]);
+
+    // Check that the original permission can be set and retrieved.
+    $event->setPermission($original_permission);
+    $this->assertEquals($original_permission, $event->getGroupContentOperationPermission($entity_type_id, $bundle_id, $operation, $ownership));
+
+    // Check that the altered permission can be set and retrieved, even though
```

Comment:
what would be the use case of that? My problem with this, is how would you expect the UI to behave?

Maybe it's better to be stricter, and allow a single permission machine name per entity/bundle/operation